data/publiccloud/terraform/gce.tf

```
@@ -90,10 +90,13 @@ resource "google_compute_instance" "openqa" {
         scopes = ["cloud-platform"]
     }
 
-    shielded_instance_config {
-        enable_secure_boot = "${var.uefi}"
-        enable_vtpm = "${var.uefi}"
-        enable_integrity_monitoring = "${var.uefi}"
+    dynamic "shielded_instance_config" {
+        for_each = "${var.uefi}" ? [ "UEFI" ] : []
```

the content of the list is irrelevant as long as it contains 1 element.

---

It looks already good to me, only if you like it

```suggestion
        for_each = ${var.uefi} ? [ "" ] : []
```

```
variable "uefi" {
    default=false
    type=bool
}
```

---

good idea, but it's not the only place where we are defining boolean variables, if I do it here I feel that I should add it in all the other occurrences of booleans..  maybe some work for another PR to improve the code or the tf files, which I think we should some time.

---

ok